controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -130,6 +144,17 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 		return response, errors.Errorf("there are %d healthy etcd pods, but %d etcd members", expectedMembers, len(knownMemberIDSet))
 	}
 
+	// If the member list view of the nodes are different, this indicates a problem in the etcd cluster like partition.
+	// Do not perform below check as its result will differ by the member list used.
+	if !inconsistentMemberList{
```

Comment:
// If the nodes are resporting a consistent member list make a further check enuring that all the etcd members 
        // have a corresponding node (so there should not be "external" etcd members). 
	if nodesResportSameMemberList{